File: primerTrimestre/java basico/3repetitivas/Ej8rep.java
/**Hacer un programa que muestre un cronometro, indicando las horas, minutos y segundos.
 *
 * @author Antonio Quesada
 */

public class Ej8rep {
  public static void main(String[] args) {
    
    //Inicializamos las variables a 0.
    int segundos = 0;
    int minutos = 0;
    int horas = 0;
    
    while(true) {
        
        //Incrementar segundos.
      if(segundos!=59) {
        segundos++;
        
        //Incrementamos los minutos cada 60 segundos
      }else if(minutos!=59) {
          minutos++;
          segundos = 0;
         
          //Ponemos los segundos a 0 e incrementamos las horas cada 60 minutos.
        }else{
          horas++;
          minutos = 0; //Ponemos a 0 los minutos y asi sucesivamente.
        }
      }
      
    //Mostramos el cronómetro
      System.out.println(horas+":"+minutos+":"+segundos);
   
  }
}

File: primerTrimestre/introduccion_programacion_python/3repetitivas_py/ej8rep.py
# Programa: ej8rep.py
# Propósito: Hacer un programa que muestre un cronometro, indicando las horas, minutos y segundos.
# Para hacer una espera en Python podemos usar el método sleep del módulo time.
#
# Autor: Antonio Quesada
# Fecha: 30/10/2019.
#
# 
#


import time

# Inicializamos las variables.
horas = 0
minutos = 0
segundos = 0

# Proceso.
# Ciclo infinito y esperamos un segundo cada iteración.
 
while True:
    print(horas,":",minutos,":",segundos)
    time.sleep(1)
    
    # pasar al siguiente segundo.
    if segundos < 59:
        segundos+=1
    else:
        segundos=0
        if minutos < 59:
            minutos+=1
        else:
            minutos=0
            horas+=1
            
            
    #ponemos el cursor al principio de la linea.
            
    print()        
    



